codersdk/deployment.go

```diff
@@ -1762,6 +1762,26 @@ type LinkConfig struct {
 	Icon   string `json:"icon" yaml:"icon"`
 }
 
+// DeploymentOptionsWithoutSecrets returns a copy of the OptionSet with secret values omitted.
+func DeploymentOptionsWithoutSecrets(set clibase.OptionSet) clibase.OptionSet {
+	cpy := make(clibase.OptionSet, 0, len(set))
+	for _, opt := range set {
+		cpyOpt := opt
+		if IsSecretDeploymentOption(cpyOpt) {
+			switch cpyOpt.Value.Type() {
+			case "string-array":
+				var empty clibase.StringArray
+				cpyOpt.Value = &empty
+			default:
+				var empty clibase.String
+				cpyOpt.Value = &empty
+			}
+		}
+		cpy = append(cpy, cpyOpt)
+	}
+	return cpy
+}
```

Comment:
This feels like it should be method of `clibase.OptionSet` instead.

Reply:
I originally had that, but the annotations are a `codersdk` constant, and cannot be imported in `clibase`.

Reply:
https://github.com/coder/coder/blob/1afa4fe47e43394a19d568e7a99088089bfe2320/cli/clibase/option_test.go#L211-L211